### PR TITLE
Issue #844 Some more javadoc fixes

### PR DESCRIPTION
**Pull request by ljacomet**
* core spi and sub packages
* Renamed `CacheAccessException` to `StoreAccessException`
* Moving classes into internal packages

### CODE REVIEW

File: CONTRIBUTING.adoc [hunk 1]
```diff
@@ -40,11 +40,11 @@ The core code of Ehcache is split in three modules:
...
 
-`Service` are created by using the Java's https://docs.oracle.com/javase/6/docs/api/java/util/ServiceLoader.html[`java.util.ServiceLoader` service-provider loading facility]. It is used to locate all `org.ehcache.core.spi.service.ServiceFactory` implementations on the classpath. These are in turn used to create `Service` instances. Each `CacheManager` uses its own `org.ehcache.core.spi.ServiceLocator` facility to locate `Service` instances, which it then in turn life cycles.
+`Service` are created by using the Java's https://docs.oracle.com/javase/6/docs/api/java/util/ServiceLoader.html[`java.util.ServiceLoader` service-provider loading facility]. It is used to locate all `org.ehcache.core.spi.service.ServiceFactory` implementations on the classpath. These are in turn used to create `Service` instances. Each `CacheManager` uses its own `org.ehcache.core.internal.service.ServiceLocator` facility to locate `Service` instances, which it then in turn life cycles.
```
Comment (chrisdennis) [inline CONTRIBUTING.adoc]: Services are created using Java's

File: core/src/main/java/org/ehcache/core/config/store/package-info.java [hunk 2]
```diff
@@ -15,6 +15,6 @@
  */
 
 /**
- * Package holding implementation of {@link org.ehcache.core.spi.cache.Store.Configuration} and related types.
+ * Package holding implementation of {@link org.ehcache.core.spi.store.Store.Configuration} and related types.
```
Comment (chrisdennis) [inline core/src/main/java/org/ehcache/core/config/store/package-info.java]: implementations

File: core/src/main/java/org/ehcache/core/exceptions/CachePassThroughException.java [hunk 3]
```diff
@@ -52,24 +52,24 @@ public CachePassThroughException(final Throwable cause) {
   /**
    * Helper method for handling runtime exceptions.
    * <P>
-   *   Throwing most as {@link CacheAccessException} except for {@code CachePassThroughException}.
+   *   Throwing most as {@link StoreAccessException} except for {@code CachePassThroughException}.
```
Comment (chrisdennis) [inline core/src/main/java/org/ehcache/core/exceptions/CachePassThroughException.java]: This class seems to be named weirdly now, given it's relationship with StoreAccessException.

File: core/src/main/java/org/ehcache/core/spi/service/ServiceFactory.java [hunk 4]
```diff
@@ -16,15 +16,30 @@
...
 /**
- * @author Alex Snaps
+ * A factory abstraction that can create {@link Service} instances.
  */
 public interface ServiceFactory<T extends Service> {
 
+  /**
+   * Creates an instance of the service using the passed in {@link ServiceCreationConfiguration}.
+   * <P>
+   *   Note that the mandatory aspect of the configuration is a service implementation concern.
```
Comment (chrisdennis) [inline core/src/main/java/org/ehcache/core/spi/service/ServiceFactory.java]: What "mandatory aspect"? Maybe "Note that a null configuration may be supported or even required by a service implementation"?

File: core/src/main/java/org/ehcache/core/spi/service/ServiceFactory.java [hunk 5]
```diff
@@ -16,15 +16,30 @@
...
+   * </P>
+   * @param configuration the creation configuration, can be {@code null} for some services
+   *
+   * @return the new service, not {@link Service#start(ServiceProvider) started}
+   */
   T create(ServiceCreationConfiguration<T> configuration);
 
+  /**
+   * Enables to query a {@code ServiceFactory} to know which {@link Service} type is produces.
```
Comment (chrisdennis) [inline core/src/main/java/org/ehcache/core/spi/service/ServiceFactory.java]: "Queries a" "it produces"

**Comment by ljacomet** [comment]
Updated up to `impl` module being covered. Does not close #844 as modules `transaction`, `107` and `management` still need to be covered.

File: core/src/main/java/org/ehcache/core/internal/store/StoreConfigurationImpl.java [hunk 6]
```diff
@@ -14,18 +14,18 @@
...
 import org.ehcache.config.ResourcePools;
 import org.ehcache.expiry.Expiry;
-import org.ehcache.core.spi.cache.Store;
+import org.ehcache.core.spi.store.Store;
 import org.ehcache.spi.serialization.Serializer;
 
 /**
- * Implementation class of the {@link Store.Configuration} as used by {@link org.ehcache.core.EhcacheManager} in order
- * to prepare {@link Store} creation.
+ * Implementation class of the {@link org.ehcache.core.spi.store.Store.Configuration} as used by
```
Comment (chrisdennis) [inline core/src/main/java/org/ehcache/core/internal/store/StoreConfigurationImpl.java]: "Implementation of..."

File: impl/src/main/java/org/ehcache/impl/config/copy/DefaultCopierConfiguration.java [hunk 7]
```diff
@@ -43,11 +60,19 @@ public DefaultCopierConfiguration(Copier<T> instance, Type type) {
...
+  /**
+   * {@inheritDoc}
+   */
   @Override
   public Class<CopyProvider> getServiceType() {
     return CopyProvider.class;
   }
 
+  /**
+   * Returns the {@link Type} for which this configuration is
```
Comment (chrisdennis) [inline impl/src/main/java/org/ehcache/impl/config/copy/DefaultCopierConfiguration.java]: `Returns the Type of this configuration`?

**Comment by chrisdennis** [comment]
Looks to be broken due to issues with rebase after the byte sizing stuff.  I'll recheck again tonight, and merge if fixed.

**Comment by ljacomet** [comment]
My bad indeed - should be fixed now

**Pull request merged**